File: scripts/seed.go
package main

import (
	"base/config"
	"base/internal/connection"
	"base/internal/data/enums"
	"base/internal/repository/querymgo"
	"base/internal/service/feature"
	"base/internal/service/group_role"
	"base/internal/service/user"
	"base/internal/utils/validator"
	"context"
	"log"
)

func permissionUser(role string) feature.Input {
	return feature.Input{
		RoleNames: []string{role},
		Name:      "User",
		KeyMenu:   "/users",
		URLView:   "/users",
		Status:    enums.StatustypeEnabled.String(),
		Api:       "/users",
		Action:    []string{"/users [post]", "/users [get]", "/users/:id [get]", "/users/:id [delete]"},
	}
}

func permissionGroupRole(role string) feature.Input {
	return feature.Input{
		RoleNames: []string{role},
		Name:      "GroupRole",
		KeyMenu:   "/group-roles",
		URLView:   "/group-roles",
		Status:    enums.StatustypeEnabled.String(),
		Api:       "/group-roles",
		Action:    []string{"/group-roles [post]", "/group-roles [get]", "/group-roles/:id [get]", "/group-roles/:id [put]", "/group-roles/:id [delete]"},
	}
}

func permissionFeature(role string) feature.Input {
	return feature.Input{
		RoleNames: []string{role},
		Name:      "Feature",
		KeyMenu:   "/features",
		URLView:   "/features",
		Status:    enums.StatustypeEnabled.String(),
		Api:       "/features",
		Action:    []string{"/features [post]", "/features [get]", "/features/:id [get]", "/features/:id [put]", "/features/:id [delete]"},
	}
}

func superAdmin() user.Input {
	return user.Input{
		Email:    "admin@admin.com",
		Phone:    "",
		Password: "Admin123@",
		Fullname: "super admin",
		Birthday: "",
	}
}

func roleAdmin(userId string) group_role.GroupRoleInput {
	return group_role.GroupRoleInput{
		Name:        "admin",
		Description: "super admin",
		Status:      enums.StatustypeEnabled.String(),
		Members:     []string{userId},
	}
}

func seedSuperAdmin(ctx context.Context, userService user.Service, groupRoleService group_role.Service, featureService feature.Service) {
	sAdmin, err := userService.Create(ctx, superAdmin())
	if err != nil {
		log.Fatal("Create super admin error:", err)
	}

	sAdminGroupRole, err := groupRoleService.Create(ctx, roleAdmin(sAdmin.ID))
	if err != nil {
		log.Fatal("Create group role error:", err)
	}

	p := permissionUser(sAdminGroupRole.Name)
	_, err = featureService.Create(ctx, p)
	if err != nil {
		log.Fatal("Create feature error:", err)
	}

	p = permissionGroupRole(sAdminGroupRole.Name)
	_, err = featureService.Create(ctx, p)
	if err != nil {
		log.Fatal("Create feature error:", err)
	}

	p = permissionFeature(sAdminGroupRole.Name)
	_, err = featureService.Create(ctx, p)
	if err != nil {
		log.Fatal("Create feature error:", err)
	}

	log.Println("✅ Tạo admin thành công.")
}

func main() {
	var ctx = context.Background()
	var cf = config.LoadEnv()
	var db = connection.ConnectDB(ctx, cf.DB)
	var userRepo = querymgo.NewUserRepo(db, "users", "usr")
	var groupRoleRepo = querymgo.NewRoleRepo(db, "group_roles", "gr")
	var featureRepo = querymgo.NewFeatureRepo(db, "features", "ft")

	var validatorService = validator.NewValidator()
	var userService = user.NewUserService(userRepo, validatorService)
	var groupRoleService = group_role.NewGroupRoleService(groupRoleRepo, validatorService)
	var featureService = feature.NewFeatureService(featureRepo, validatorService)

	seedSuperAdmin(ctx, userService, groupRoleService, featureService)
}

File: scripts/seed.js

db = db.getSiblingDB("golang_test");

const time = Date.now()

// 1. Insert user
const user = {
  _id:"usr_fb4a15c6-3fe9-488a-b853-bc869d37eb45",
  username: "admin",
  email: "admin@admin.com",
  phone: "",
  fullname: "super admin",
  birthday: "",
  password: "$2a$10$Pnmvw2eudCsIhlUtL6ojwuoJbPciniUNMbwHnhHCTMzbOxDtIyji2",
  ctime: time,
  mtime: time,
  dtime: 0
};
db.users.insertOne(user);

const groupRole = {
  _id:"gr_67a17d82-1b0a-4412-b4f3-0b0709de586c",
  name:  "admin",
  description: "super admin",
  status:  "status_type_enabled",
  member_ids: ["usr_fb4a15c6-3fe9-488a-b853-bc869d37eb45"],
  ctime: time,
  mtime: time,
  dtime: 0
};
db.group_roles.insertOne(groupRole);

const feature = {
  _id: "ft_435816b9-be0d-4ece-8ea6-9a9917548650",
  name: "Feature",
  key_menu:  "/features",
  url_view:  "/features",
  api:  "/features",
  action: ["/features [post]", "/features [get]", "/features/:id [get]", "/features/:id [put]", "/features/:id [delete]"],
  status: "status_type_enabled",
  role_names: ["admin"],
  ctime:time,
  mtime: time,
  dtime: 0
};
db.features.insertOne(feature);

const roleF = {
  _id: "ft_05982bf7-cd23-4723-a1ed-8e291204c272",
  name: "GroupRole",
  key_menu:  "/group-roles",
  url_view:  "/group-roles",
  api:  "/group-roles",
  action: ["/group-roles [post]", "/group-roles [get]", "/group-roles/:id [get]", "/group-roles/:id [put]", "/group-roles/:id [delete]"],
  status: "status_type_enabled",
  role_names: ["admin"],
  ctime:time,
  mtime: time,
  dtime: 0
};
db.features.insertOne(roleF);

const uF = {
  _id: "ft_7a6d63dd-23f8-4f29-8302-0590a9bc3864",
  name: "User",
  key_menu:   "/users",
  url_view:   "/users",
  api:   "/users",
  action: ["/users [post]", "/users [get]", "/users/:id [get]", "/users/:id [delete]"],
  status: "status_type_enabled",
  role_names: ["admin"],
  ctime:time,
  mtime: time,
  dtime: 0
};
db.features.insertOne(uF);
